nodejs/eks/nodegroup.ts

```diff
@@ -1728,37 +1748,55 @@ function createMNGCustomLaunchTemplate(
     parent: pulumi.Resource,
     provider?: pulumi.ProviderResource,
 ): aws.ec2.LaunchTemplate {
-    const kubeletExtraArgs = args.kubeletExtraArgs ? args.kubeletExtraArgs.split(" ") : [];
-    let bootstrapExtraArgs = args.bootstrapExtraArgs ? " " + args.bootstrapExtraArgs : "";
-
-    if (kubeletExtraArgs.length === 1) {
-        // For backward compatibility with previous versions of this package, don't wrap a single argument with `''`.
-        bootstrapExtraArgs += ` --kubelet-extra-args ${kubeletExtraArgs[0]}`;
-    } else if (kubeletExtraArgs.length > 1) {
-        bootstrapExtraArgs += ` --kubelet-extra-args '${kubeletExtraArgs.join(" ")}'`;
-    }
+    let userData;
 
-    const userdata = pulumi.all([core.cluster.name, core.cluster.endpoint, core.cluster.certificateAuthority.data, args.clusterName]).apply(([clusterName, clusterEndpoint, clusterCertAuthority, argsClusterName]) => {
-        return `#!/bin/bash
+    // If the user specifies either kubeletExtraArgs or bootstrapExtraArgs, we need to create a base64 encoded user data script.
+    if (args.kubeletExtraArgs || args.bootstrapExtraArgs) {
+        const kubeletExtraArgs = args.kubeletExtraArgs ? args.kubeletExtraArgs.split(" ") : [];
```

Comment:
Unrelated - I think you want to strip this in case you get something like `" foo "`.